decode_test.go

```diff
@@ -145,11 +161,28 @@ func TestDecode(t *testing.T) {
 					}
 
 					results[fmt] = val
-					if !reflect.DeepEqual(expVal, val) {
-						t.Logf("Expected: %#v\n", expVal)
-						t.Logf("Received: %#v\n", val)
-						t.Fail()
+					require.Equal(t, expVal, val)
+				})
+				subtest(t, FormatNames[fmt], func(t *testing.T) {
```

Comment:
i do not get this new subtest, it seems the 1. subtests runs the original cases (without TestCustomAnnotiation)
while the 2. subtest runs all the cases, it seems this way we run some test cases 2 times.

Reply:
One calls Unmarshal the other the custom Unmarshal